pkg/cmd/cluster-policy-controller/cmd.go

```diff
@@ -1,115 +1,300 @@
+	"github.com/openshift/library-go/pkg/config/client"
+	"github.com/openshift/library-go/pkg/config/configdefaults"
+	"github.com/openshift/library-go/pkg/controller/controllercmd"
+	"github.com/openshift/library-go/pkg/controller/fileobserver"
+	"github.com/openshift/library-go/pkg/crypto"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/serviceability"
+
+	// for metrics
+	_ "github.com/openshift/library-go/pkg/controller/metrics"
+
+	clusterpolicyversion "github.com/openshift/cluster-policy-controller/pkg/version"
 )
 
 const RecommendedStartControllerName = "cluster-policy-controller"
 
-type ClusterPolicyController struct {
-	ConfigFilePath string
-	// KubeConfigFile points to a kubeconfig file if you don't want to use the in cluster config
-	KubeConfigFile string
+func NewClusterPolicyControllerCommand(name string) *cobra.Command {
+	cmd := NewClusterPolicyControllerCommandConfig("cluster-policy-controller", clusterpolicyversion.Get(), RunClusterPolicyController).
+		NewCommandWithContext(context.TODO())
+	cmd.Use = name
+	cmd.Short = "Start the cluster policy controller"
+
+	return cmd
+}
+
+// ClusterPolicyControllerCommandConfig holds values required to construct a command to run.
+// copied from controllercmd.ControllerCommandConfig
+type ClusterPolicyControllerCommandConfig struct {
+	componentName string
+	startFunc     controllercmd.StartFunc
+	version       version.Info
+
+	basicFlags *controllercmd.ControllerFlags
 
-	Output io.Writer
+	// DisableServing disables serving metrics, debug and health checks and so on.
+	DisableServing bool
+
+	// DisableLeaderElection allows leader election to be suspended
+	DisableLeaderElection bool
 }
 
-func NewClusterPolicyControllerCommand(name string, out, errout io.Writer, stopCh <-chan struct{}) *cobra.Command {
-	options := &ClusterPolicyController{Output: out}
+// NewClusterPolicyControllerCommandConfig returns a new ClusterPolicyControllerCommandConfig which can be used to wire up all the boiler plate of a controller
+func NewClusterPolicyControllerCommandConfig(componentName string, version version.Info, startFunc controllercmd.StartFunc) *ClusterPolicyControllerCommandConfig {
+	return &ClusterPolicyControllerCommandConfig{
+		startFunc:     startFunc,
+		componentName: componentName,
+		version:       version,
+
+		basicFlags: controllercmd.NewControllerFlags(),
 
+		DisableServing:        false,
+		DisableLeaderElection: false,
+	}
+}
+
+// NewCommandWithContext returns a new command that a caller must set the Use and Descriptions on.  It wires default log, profiling,
+// leader election and other "normal" behaviors.
+// The context passed will be passed down to controller loops and observers and cancelled on SIGTERM and SIGINT signals.
+// copied from controllercmd.ControllerCommandConfig
+func (c *ClusterPolicyControllerCommandConfig) NewCommandWithContext(ctx context.Context) *cobra.Command {
```

Comment:
This is copy pasted logic for now, since we need a custom logic in StartController and access to the internal variables.

I will make a PR against library-go so this could be customizable and we could avoid copy paste